### PR TITLE
Resolves a null pointer exception when returning a Dictionary containing known keys

### LINKED ISSUE #2368: Potential problems with 6.3.0
Running a .net 6.0 C# Web API.

After upgrading Swashbuckle.AspNetCore  from 6.2.3 to 6.3.0,
the API will not start, and gives the following error message:

>  Swashbuckle.AspNetCore.SwaggerGen.SwaggerGeneratorException: Failed to generate Operation for action(ActionName) . See inner exception
 ---> Swashbuckle.AspNetCore.SwaggerGen.SwaggerGeneratorException: Failed to generate schema for type(ActionType). See inner exception
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchemaForMember(Type modelType, SchemaRepository schemaRepository, MemberInfo memberInfo, DataProperty dataProperty)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.CreateObjectSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.<>c__DisplayClass10_0.<GenerateConcreteSchema>b__3()
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateReferencedSchema(DataContract dataContract, SchemaRepository schemaRepository, Func`1 definitionFactory)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateConcreteSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchemaForMember(Type modelType, SchemaRepository schemaRepository, MemberInfo memberInfo, DataProperty dataProperty)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.CreateObjectSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.<>c__DisplayClass10_0.<GenerateConcreteSchema>b__3()
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateReferencedSchema(DataContract dataContract, SchemaRepository schemaRepository, Func`1 definitionFactory)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateConcreteSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchemaForType(Type modelType, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchema(Type modelType, SchemaRepository schemaRepository, MemberInfo memberInfo, ParameterInfo parameterInfo, ApiParameterRouteInfo routeInfo)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.CreateArraySchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.<>c__DisplayClass10_0.<GenerateConcreteSchema>b__1()
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateConcreteSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchemaForMember(Type modelType, SchemaRepository schemaRepository, MemberInfo memberInfo, DataProperty dataProperty)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.CreateObjectSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.<>c__DisplayClass10_0.<GenerateConcreteSchema>b__3()
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateReferencedSchema(DataContract dataContract, SchemaRepository schemaRepository, Func`1 definitionFactory)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateConcreteSchema(DataContract dataContract, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchemaForType(Type modelType, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchema(Type modelType, SchemaRepository schemaRepository, MemberInfo memberInfo, ParameterInfo parameterInfo, ApiParameterRouteInfo routeInfo)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateSchema(Type type, SchemaRepository schemaRepository, PropertyInfo propertyInfo, ParameterInfo parameterInfo, ApiParameterRouteInfo routeInfo)
   --- End of inner exception stack trace ---
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateSchema(Type type, SchemaRepository schemaRepository, PropertyInfo propertyInfo, ParameterInfo parameterInfo, ApiParameterRouteInfo routeInfo)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreateResponseMediaType(ModelMetadata modelMetadata, SchemaRepository schemaRespository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.<>c__DisplayClass19_0.<GenerateResponse>b__2(String contentType)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateResponse(ApiDescription apiDescription, SchemaRepository schemaRepository, String statusCode, ApiResponseType apiResponseType)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateResponses(ApiDescription apiDescription, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperation(ApiDescription apiDescription, SchemaRepository schemaRepository)
   --- End of inner exception stack trace ---
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperation(ApiDescription apiDescription, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperations(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePaths(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)a
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)

 by downgrading back to 6.2.3 it started working again.

I also have the Swashbuckle.AspNetCore.Newtonsoft installed - but any permutation of version 6.3.0 gives the same error message.

### PR DESCRIPTION
**Pull request by Eneuman**
This PR fixes a null pointer exception when having a API returning av Dictionary containing known keys
(ie a Dictionary<Enum, string>).

The problem was that the code tried to set `schema.AdditionalProperties.Nullable` without first checking if `schema.AdditionalPropertiesAllowed` was true. This resulted in a null point exception since `schema.AdditionalProperties` is null when `schema.AdditionalPropertiesAllowed` is false.

Fixes #2368

### CODE REVIEW

**Comment by jelical** [comment]
Hello, there @domaindrivendev :) Any news on this?

**Review by JakubHolovsky — APPROVED** [review]

**Comment by sergey-litvinov** [comment]
@domaindrivendev is there any update on this PR and related issue? i'm having the same problem related to Dictionary and NRE during schema generation

**Comment by Eneuman** [comment]
@domaindrivendev ping

**Comment by tomhreb** [comment]
Hi @domaindrivendev, @JakubHolovsky -- any news on this please?

**Comment by plutora-steve-g** [comment]
The issue is still present in 6.4.0

**Comment by HugoKampsSB** [comment]
A workaround which helped for me was changing my response type to  `List<KeyValuePair<A,B>>` instead of `Dictionary<A,B>`. When building my response I only had to add `.ToList()` after my Dictionary initialization, and this fixed the issue for now.

**Comment by asmcpluss** [comment]
Any updates here?

**Comment by jason-daly** [comment]
I just tripped over this issue after updating to 6.4.0. Would be nice to see it fixed.

**Comment by plachor** [comment]
Looking forward fix as well.

**Comment by fschick** [comment]
Same issue here. Can't update to 6.4.0.

**Comment by JakubHolovsky** [comment]
> Hi @domaindrivendev, @JakubHolovsky -- any news on this please?

I am not the maintainer but trying to get all the attention to this so it finally gets merged. Stuck on older version here as well.

**Comment by sergei-lobanov** [comment]
@domaindrivendev hello, could you please approve this PR? We are all waiting months...

**Comment by hecerinc** [comment]
@domaindrivendev could we possibly get a release with this fix in? Thank you!

**Comment by angelaki** [comment]
Waiting for this fix, too. Soon deploy would be awesome!

**Comment by sguryev** [comment]
We have made Dictionary enum key nullable `Dictionary<Enum?, Model>` and SwaggerUI looks this way now
image
Waiting for the release 🤞🏼

**Comment by oyzar** [comment]
When will a version with this release be fixed?

**Pull request merged**
